operator/webhook/introute/sync.py

```diff
@@ -163,33 +200,69 @@ def _spring_app_config_env_var(parent) -> Mapping[str, str]:
     }
 
 
-def _get_java_jdk_options(parent) -> Optional[Mapping[str, str]]:
+def _get_tls_options(parent) -> Optional[List[Mapping[str, str]]]:
     tls_config = parent["spec"].get("tls")
 
     if not tls_config:
         return None
 
-    tls_type = tls_config['type']
-    assert tls_type in ["jks", "pkcs12"], \
-        f"({tls_type}) is not a supported TLS type. Supported types: ('jks', 'pkcs12')"
-
-    truststore_password = "changeit" if tls_type == "jks" else ""
-    jdk_options = f"-Djavax.net.ssl.trustStore={str(Path(TRUSTSTORE_PATH, tls_config['key']))} -Djavax.net.ssl.trustStorePassword={truststore_password} -Djavax.net.ssl.trustStoreType={tls_type.upper()}"
+    tls_env_vars = []
+    if 'truststore' in tls_config:
+        tls_type = tls_config['truststore']['type']
+        _validate_tls_type(tls_type)
+        truststore_password = "changeit" if tls_type == "jks" else ""
+        truststore_vars = [
+            {
+                "name": "SERVER_SSL_TRUSTSTORE",
+                "value": str(Path(TRUSTSTORE_PATH, tls_config['truststore']['key']))
+            },
+            {
+                "name": "SERVER_SSL_TRUSTSTOREPASSWORD",
+                "value": truststore_password
+            },
+            {
+                "name": "SERVER_SSL_TRUSTSTORETYPE",
+                "value": tls_type.upper()
+            }
+        ]
+        tls_env_vars.extend(truststore_vars)
 
-    return {
-        "name": "JDK_JAVA_OPTIONS",
-        "value": jdk_options,
-    }
+    if 'keystore' in tls_config:
+        tls_type = tls_config['keystore']['type']
+        _validate_tls_type(tls_type)
+        keystore_vars = [
+            {
+                "name": "SERVER_PORT",
+                "value": 443
+            },
+            {
+                "name": "SERVER_SSL_KEYSTORE",
+                "value": str(Path(KEYSTORE_PATH, tls_config['keystore']['key']))
+            },
+            {
+                "name": "SERVER_SSL_KEYSTORETYPE",
+                "value": tls_type.upper()
+            },
+            {
+                "name": "SERVER_SSL_KEYALIAS",
+                "value": "certificate"
+            }
+        ]
+        tls_env_vars.extend(keystore_vars)
+    return tls_env_vars
 
+def _validate_tls_type(tls_type):
+    assert tls_type in ["jks", "pkcs12"], \
+        f"({tls_type}) is not a supported TLS type. Supported types: ('jks', 'pkcs12')"
```

Comment:
:memo: I think this validation logic is already enforced by the `enum` specification in the CRD.

Reply:
I removed this check and the associated tests.